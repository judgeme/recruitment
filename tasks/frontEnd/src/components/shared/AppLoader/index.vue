<script setup lang="ts">
import { defineOptions } from 'vue';

// set display name in dev-tools
defineOptions({
  name: 'AppLoader'
});
</script>
<template>
	<span
		class="
			loader w-12 h-12 rounded-full 
			inline-block relative border-solid border-[3px]
			border-t-primary border-r-primary 
			border-b-transparent border-l-transparent
			after:absolute before:absolute 
			after:inset-0 before:inset-0
			after:m-auto before:m-auto 
			after:border-solid after:border-[3px]
			before:border-solid before:border-[3px] 
			after:border-t-transparent after:border-r-transparent
			after:border-b-primary after:border-l-primary
			after:w-10 after:h-10 after:rounded-full before:rounded-full
			before:w-8 before:h-8 before:border-t-primary
			before:border-r-primary
			before:border-b-transparent before:border-l-transparent
			animate-spin
		"
	>
	</span>
</template>
<style lang="css" scoped>
.loader::after,
.loader::before {
	content: '';
	animation: rotationBack 0.5s linear infinite;
	transform-origin: center center;
}
.loader::before {
	animation: rotation 1.5s linear infinite;
}

@keyframes rotation {
	0% {
		transform: rotate(0deg);
	}
	100% {
		transform: rotate(360deg);
	}
}
@keyframes rotationBack {
	0% {
		transform: rotate(0deg);
	}
	100% {
		transform: rotate(-360deg);
	}
}
</style>
