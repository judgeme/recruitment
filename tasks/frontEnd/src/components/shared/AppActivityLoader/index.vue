<script setup lang="ts">
import { defineOptions, withDefaults, defineProps } from 'vue';

import AppLoader from '@/components/shared/AppLoader/index.vue';

// set display name in dev-tools
defineOptions({
  name: 'AppActivityLoader'
});

withDefaults(defineProps<{
  loading?: boolean
}>(), {
  loading: false
});
</script>
<template>
  <div class="relative w-full overflow-hidden min-h-50">
    <div v-if="loading" class="absolute inset-0 flex justify-center items-center z-10 bg-white/55">
      <app-loader />
    </div>
    <slot />
  </div>
</template>
