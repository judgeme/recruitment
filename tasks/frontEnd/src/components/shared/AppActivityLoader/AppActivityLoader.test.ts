import { mount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';
import AppActivityLoader from '@/components/shared/AppActivityLoader/index.vue';
import AppLoader from '@/components/shared/AppLoader/index.vue';

describe('AppActivityLoader.vue', () => {
  it('renders correctly and matches snapshot when loading is false', () => {
    const wrapper = mount(AppActivityLoader, {
      props: { loading: false },
      slots: {
        default: '<p>Some default content here</p>',
      },
    });

    // Check if the component is rendered
    expect(wrapper.exists()).toBe(true);

    // Check if the AppLoader is not rendered
    expect(wrapper.findComponent(AppLoader).exists()).toBe(false);

    // Snapshot test
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly and matches snapshot when loading is true', () => {
    const wrapper = mount(AppActivityLoader, {
      props: { loading: true },
      slots: {
        default: '<p>Some default content here</p>',
      },
    });

    // Check if the component is rendered
    expect(wrapper.exists()).toBe(true);

    // Check if the AppLoader is rendered
    expect(wrapper.findComponent(AppLoader).exists()).toBe(true);

    // Snapshot test
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders slot content correctly', () => {
    const wrapper = mount(AppActivityLoader, {
      props: { loading: false },
      slots: {
        default: '<p>Some default content here</p>',
      },
    });

    // Check if the slot content is rendered
    expect(wrapper.html()).toContain('Some default content here');
  });
});
