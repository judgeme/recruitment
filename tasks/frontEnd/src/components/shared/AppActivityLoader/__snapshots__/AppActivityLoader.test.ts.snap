// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppActivityLoader.vue > renders correctly and matches snapshot when loading is false 1`] = `
"<div class="relative w-full overflow-hidden min-h-10">
  <!--v-if-->
  <p>Some default content here</p>
</div>"
`;

exports[`AppActivityLoader.vue > renders correctly and matches snapshot when loading is true 1`] = `
"<div class="relative w-full overflow-hidden min-h-10">
  <div class="absolute inset-0 flex justify-center items-center z-10 bg-white/55"><span data-v-9e3594fe="" class="loader w-12 h-12 rounded-full inline-block relative border-solid border-[3px] border-t-primary border-r-primary border-b-transparent border-l-transparent after:absolute before:absolute after:inset-0 before:inset-0 after:m-auto before:m-auto after:border-solid after:border-[3px] before:border-solid before:border-[3px] after:border-t-transparent after:border-r-transparent after:border-b-primary after:border-l-primary after:w-10 after:h-10 after:rounded-full before:rounded-full before:w-8 before:h-8 before:border-t-primary before:border-r-primary before:border-b-transparent before:border-l-transparent animate-spin"></span></div>
  <p>Some default content here</p>
</div>"
`;
